runtime/drivers/druid/olap.go

```diff
@@ -43,6 +43,12 @@ func (c *connection) Execute(ctx context.Context, stmt *drivers.Statement) (*dri
 		return nil, prepared.Close()
 	}
 
+	var cancelFunc context.CancelFunc
+	if stmt.ExecutionTimeout != 0 {
+		ctx, cancelFunc = context.WithTimeout(ctx, stmt.ExecutionTimeout)
+		_ = cancelFunc
+	}
```

Comment:
This and setting it in the result is not needed. Firstly, the `ctx` is used only to resolve the query, not for reading the result, so there it doesn't make a difference to delay its cancellation until after the rows have been read. Secondly, the execution timeout is for execution, not for reading the results (that would be covered by the API-level timeout)

Reply:
So can just do 
```go
if stmt.ExecutionTimeout != 0 {
	cctx, cancelFunc := context.WithTimeout(ctx, stmt.ExecutionTimeout)
	defer cancelFunc()
	ctx = cctx
}
```

Reply:
Nope, once the timeout is cancelled the rows cannot be read.

Reply:
For example:
```
func Execute() {
        ...
	if stmt.ExecutionTimeout != 0 {
		ctx, cancel = context.WithTimeout(ctx, stmt.ExecutionTimeout)
		cancel()
	}

	rows, err := conn.QueryxContext(ctx, stmt.Query, stmt.Args...)
        ...
        return rows // the context is canceled at this point
}

func processing() {
        rows, err := Execute()
        ...
        if rows.next() {
          err = rows.Scan(&timeGrainString) // we get context-cancelled error here
        }
}


```

Reply:
Okay, I see. Slightly odd behavior from the library there, please add a comment before `context.WithTimeout` that the context is not released because it will also be used for reading rows.

Also, instead of the `CancelFunc` on `Result`, can't you just use the `SetCleanupFunc` that's already there? Instead of just passing `release` to it, you would pass a func that calls both `release` and `cancel`